middleware/src/configuration/configuration.go

```diff
@@ -2,7 +2,37 @@
 // values to the Middleware.
 package configuration
 
+// Args has the same fields as the `Configuration` struct, but the fields in
+// `Args` are public. The struct is used as parameter to the `NewConfiguration()`
+// factory function. The struct needs public fields to be settable the `main`
+// package. However the `Configuration` can't have public fields, because the
+// fields should be **READ ONLY** while the Middleware is running. Fields should
+// only be accessible via defined Getter functions.
+//
+// Note: Go does not support named arguments for functions. While passing the
+// 			 arguments to a function would work, it would be rather easy to
+// 			 mistakenly switch e.g. two string parameters ending up with an invalid
+// 			 middleware configuration. Using the `Args` helps, because a Go struct
+//       can be initialized with named fields.
```

Comment:
Mix of tab & space indent?

Reply:
yeah probably. Will remove indent, as it's not really needed.